### ISSUE TITLE
Rule proposal: `prefer-map-has`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1329]
When checking existence, should prefer `Map#has()` over `Map#get()`.

## Fail

```js
if (map.get(foo));
```

## Pass

```js
if (map.has(foo));
```

I think better ship this with [`prefer-map`](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1285) together, so user can notice this problem when we fix `object.foo` to `map.get(foo)` in `prefer-map`.

**Comment by sindresorhus** [comment]
Accepted.

**Comment by fregante** [comment]
It makes sense, but what if map is specifically full of booleans?

**Comment by fisker** [comment]
The `unicorn/prefer-array-some` is similar, the old `array.find()` maybe skipping falsely values, so it's fine to add this rule, just don't auto-fix.

**Comment by axetroy** [comment]
I think it should not be called this name. Because it is not exclusive to `Map`

There are several common objects in JavaScript that also have get/has methods.

`{Map,WeakMap,FormData,URLSearchParams.Headers}#{get,has}`

**Comment by sindresorhus** [comment]
Yeah, maybe `prefer-has-check`

**Comment by fisker** [comment]
> It makes sense, but what if map is specifically full of booleans?

I feel I was wrong, it's common to store falsely values in a `Map`, especially `0`/`false`, and I don't  think we have a good way to make sure it's just checking existence. I'm sure there will be false positives,

Maybe I should close?

**Comment by axetroy** [comment]
> it's common to store falsely values in a Map, especially 0/false, and I don't think we have a good way to make sure it's just checking existence.

Agreed

**Comment by fregante** [comment]
Are type-aware issues not allowed in Unicorn? This rule can still work if it excludes Boolean sets and maps
